lib/modules/manager/asdf/upgradeable-tooling.ts

```diff
@@ -0,0 +1,260 @@
+        continue;
+      }
+      return {
+        datasource: GithubReleasesDatasource.id,
+        packageName: `adoptium/temurin${majorVersion}-binaries`,
+        versioning: hermitVersioning.id,
+        extractVersion: '^adoptopenjdk-(?<version>\\S+)',
+      };
+    }
+    return undefined;
+  },
+  julia: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'JuliaLang/julia',
+    versioning: semverVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  just: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'casey/just',
+    versioning: semverVersioning.id,
+  },
+  kotlin: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'JetBrains/kotlin',
+    versioning: semverVersioning.id,
+    extractVersion: '^(Kotlin |v)(?<version>\\S+)',
+  },
+  kustomize: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'kubernetes-sigs/kustomize',
+    versioning: semverVersioning.id,
+    extractVersion: '^kustomize/v(?<version>\\S+)',
+  },
+  lua: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'lua/lua',
+    versioning: semverVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  nim: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'nim-lang/Nim',
+    versioning: semverVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  nodejs: {
+    depName: 'node',
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'nodejs/node',
+    versioning: nodeVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  ocaml: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'ocaml/ocaml',
+    versioning: semverVersioning.id,
+  },
+  perl: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'Perl/perl5',
+    versioning: semverVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  php: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'php/php-src',
+    versioning: semverVersioning.id,
+    extractVersion: '^php-(?<version>\\S+)',
+  },
+  python: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'python/cpython',
+    versioning: semverVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  ruby: {
+    datasource: 'ruby-version',
+    packageName: 'ruby-version',
+    versioning: semverVersioning.id,
+  },
+  rust: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'rust-lang/rust',
+    versioning: semverVersioning.id,
+  },
+  scala: (version) => {
+    if (version.startsWith('2')) {
+      return {
+        datasource: GithubTagsDatasource.id,
+        packageName: 'scala/scala',
+        versioning: semverVersioning.id,
+        extractVersion: '^v(?<version>\\S+)',
+      };
+    }
+    if (version.startsWith('3')) {
+      return {
+        datasource: GithubTagsDatasource.id,
+        packageName: 'lampepfl/dotty',
+        versioning: semverVersioning.id,
+      };
+    }
```

Comment:
why does v3 uses a different datasource?

Reply:
@kachick What is the reason for the two sources in the asdf config preset?

Reply:
As far as I know, they have different repository for scala 2 and scala 3.
Official doc has the reference as https://github.com/scala/docs.scala-lang/blob/da2e086f5d684f850243af01fe98097a0ebc605f/scala3/reference/README.md?plain=1#L1-L6

However I'm not a scala user, it might be an misunderstanding 🤔

Reply:
The `scala/scala` repo does not have any major version 3 tags: https://github.com/scala/scala/tags

So I believe that is correct.